### PR TITLE
Plugins: Fix don't set Content-Type header if status is 204 for call resource

### PR DESCRIPTION
**Pull request by sd2k**
Grafana's HTTPServer ensures that the Content-Type header is always set
in the response to a CallResource call, but when the status code is
204 No Content this shouldn't be done; the body should be empty and no
Content-Type header should be set.

We ran into this in the Grafana ML plugin where we were sending an empty
response with status 204, but the frontend client saw that the content
type was JSON and tried to parse it, resulting in an error that made it
to the JS console.

### CODE REVIEW

**Review by marefr — APPROVED** [review]
LGTM

**Comment by sd2k** [comment]
Quick ping: this has 1 approval, does it need more before being merged?

**Comment by marefr** [comment]
> Quick ping: this has 1 approval, does it need more before being merged?

@sd2k no please go ahead, but would maybe suggest syncing with latest main before merging since it's an old branch.

**Comment by marefr** [comment]
In addition you need to [assign a milestone](https://github.com/grafana/grafana/blob/main/contribute/merge-pull-request.md#assign-a-milestone), [decide if include in changelog or not](https://github.com/grafana/grafana/blob/main/contribute/merge-pull-request.md#include-in-changelog-and-release-notes) and [whether it needs backporting](https://github.com/grafana/grafana/blob/main/contribute/merge-pull-request.md#should-the-pull-request-be-backported).

**Comment by sd2k** [comment]
> > Quick ping: this has 1 approval, does it need more before being merged?
> 
> @sd2k no please go ahead, but would maybe suggest syncing with latest main before merging since it's an old branch.

Thanks @marefr, I don't think I have permissions to merge in grafana/grafana though! I'll do the milestone/changelog/backport stuff now.

**Pull request merged**
